src/VisualStudio/Core/Def/SampleToolWindow/SampleToolboxUserControl.xaml.cs

```diff
@@ -0,0 +1,460 @@
+                }
+            }
+        }
+
+        private void ExpandAll(object sender, RoutedEventArgs e)
+        {
+            var documentSymbols = new List<DocSymbol>();
+            foreach (var item in (List<DocSymbol>)symbolTree.ItemsSource)
+            {
+                documentSymbols.Add(ExpandAllNodes(item));
+            }
+
+            symbolTree.ItemsSource = documentSymbols;
+        }
+
+        private DocSymbol ExpandAllNodes(DocSymbol treeItem)
+        {
+            treeItem.IsExpanded = true;
+            foreach (var childItem in treeItem.Children.OfType<DocSymbol>())
+            {
+                ExpandAllNodes(childItem);
+            }
+
+            return treeItem;
+        }
+
+        private void CollapseAll(object sender, RoutedEventArgs e)
+        {
+            var documentSymbols = new List<DocSymbol>();
+            foreach (var item in (List<DocSymbol>)symbolTree.ItemsSource)
+            {
+                documentSymbols.Add(CollapseAllNodes(item));
+            }
+
+            symbolTree.ItemsSource = documentSymbols;
+        }
+
+        private DocSymbol CollapseAllNodes(DocSymbol treeItem)
+        {
+            treeItem.IsExpanded = false;
+            foreach (var childItem in treeItem.Children.OfType<DocSymbol>())
+            {
+                CollapseAllNodes(childItem);
+            }
+
+            return treeItem;
+        }
+
+        private void RemoveText(object sender, EventArgs e)
+        {
+            if (searchBox.Text == "Search Document Outline")
+            {
+                searchBox.Text = "";
+            }
+        }
+
+        private void AddText(object sender, EventArgs e)
+        {
+            if (string.IsNullOrWhiteSpace(searchBox.Text))
+                searchBox.Text = "Search Document Outline";
+        }
+
+        private void Search(object sender, EventArgs e)
+        {
+            if (searchBox.Text == "Search Document Outline" || string.IsNullOrWhiteSpace(searchBox.Text))
+            {
+                if (this.originalTree is not null)
+                {
+                    symbolTree.ItemsSource = this.originalTree;
+                }
+            }
+            else
+            {
+                var documentSymbols = new List<DocSymbol>();
+                if (this.originalTree is not null)
+                {
+                    foreach (var item in this.originalTree)
+                    {
+                        if (DocumentOutlineHelper.SearchNodeTree(item, searchBox.Text))
+                        {
+                            documentSymbols.Add(item);
+                        }
+                    }
+
+                    symbolTree.ItemsSource = documentSymbols;
+                }
+            }
+        }
+
+        private void SortByName(object sender, EventArgs e)
+        {
+            var items = this.originalTree;
+            var documentSymbols = items.OrderBy(x => x.Name).ToList();
```

Comment:
I would suggest refactoring this code so we move the `OrderBy` in the following three functions into `DocumentOutlineHelper.Sort` itself

Reply:
Done